sympy/stats/joint_rv_types.py

```diff
@@ -405,15 +405,17 @@ class GeneralizedMultivariateLogGammaDistribution(JointDistribution):
     def check(self, omega, v, l, mu):
         _value_check(omega.is_square, "the matrix must be square")
         for val in omega.values():
-            _value_check((val >= 0) != False and (val <= 1) != False,
+            _value_check((val >= 0, val <= 1),
                 "all values in matrix must be between 0 and 1(both inclusive).")
         _value_check(omega.diagonal().equals(ones(1, omega.shape[0])),
                         "all the elements of diagonal should be 1.")
-        _value_check((v > 0) != False, "v must be positive")
+        _value_check((v > 0), "v must be positive")
         for lk in l:
-            _value_check((lk > 0) != False, "lamda must be a positive vector.")
+            _value_check((lk > 0), "lamda must be a positive vector.")
         for muk in mu:
-            _value_check((muk > 0) != False, "mu must be a positive vector.")
+            _value_check((muk > 0), "mu must be a positive vector.")
+        _value_check(len(l) > 1,"the distribution should have at least"
+                                " two random variables.")
         _value_check(omega.shape == (len(l), len(mu)),
                         "lamda, mu should be of same length and omega should "
                         " be of shape (length of lamda, length of mu)")
```

Comment:
The message does not correlate completely with what it checks.
This will not necessarily raise if `len(l) != len(mu)`.

Reply:
Here is a brief explanation behind this check,

Let us say there are `s1`, `s2`, and `l1`, `l2`, (`s1`, `s2`) denotes `omega.shape` and (`l1`,` l2`) denote `(len(l), len(mu))`. The following 4 cases arise,

1. s1 = s2 and l1 = l2 -> will raise exception only if, (s1, s2) != (l1 = l2).
2. s1 != s2 and l1 != l2 -> will not arise, because, check for `omega.is_square` will be executed first.
3. s1 = s2 and l1 != l2 -> will always raise exception as (s1, s2) != (l1, l2)
4. s1 != s2 and l1 != l2 -> same as 2.

[This test](https://github.com/sympy/sympy/pull/16825/files#diff-66766e843fd2f704b876eef8331b7ff3R104) addresses, `len(l) != len(mu)` and it is passing currently.
Please correct me if there is something wrong in my explanation.

Reply:
That's right. The context of `omega.is_square` was not visible here.
And, it would be a different matter if the exception should be split to have a single failure message each.

Reply:
I just wanted to wrap up the check method quickly while implementing this distribution so I just wrote a compressed _value_check. I will split the check if you say but personally I think we can go with this check.